common/src/main/java/com/google/auto/common/BasicAnnotationProcessor.java

```diff
@@ -505,74 +525,402 @@ private ImmutableSetMultimap<Class<? extends Annotation>, Element> toClassKeyedM
     }
   }
 
+  /* ********************************************************************************* */
+  /* Element blue prints ************************************************************* */
+  /* ********************************************************************************* */
+
   /**
-   * A package or type name.
+   * An {@link ElementBluePrint} for an annotated element.
    *
-   * <p>It's unfortunate that we have to track types and packages separately, but since there are
+   * <p>Instead of saving elements, an {@code ElementBluePrint} is saved since there is no guarantee
+   * that any particular element will always be represented by the same object. (Reference: {@link
+   * Element}) For example, Eclipse compiler uses different Element instances per round.
+   */
+  private abstract static class ElementBluePrint {
+    public final String toString;
```

Comment:
I think this could be `private`? Or at least not `public`.

Reply:
You are right. There is no need. Thank you for this.